app/frontend/src/components/views/project-board-view/ProjectBoardView.tsx

```diff
@@ -0,0 +1,242 @@
+import TaskInput from 'components/UI/task-input/TaskInput';
+import { useStore } from 'effector-react';
+import { DragDropContext, Draggable, Droppable, DropResult } from 'react-beautiful-dnd';
+import { Outlet } from 'react-router';
+import { $selectedProject, taskMoved } from 'store/projects';
+import { $tasks, changeTaskStatusFx, createTaskFx, taskSelected } from 'store/tasks';
+import styled from 'styled-components';
+import { Status, Task } from 'types/entities';
+import CardLink from '../project-list-view/components/CardLink';
+import getTaskTitle from 'helpers/get-task-title';
+import { ChangeTaskStatusPayload } from 'types/transport/requests/task/change-task-status';
+import prepareTaskContent from 'helpers/prepare-task-content';
+import { CreateTaskMessagePayload } from 'types/transport/requests/task/create-task';
+
+
+const UNSORTED_COLUMN_ID = 'unsorted-column';
+
+/**
+ * Props of the component
+ */
+interface Props { }
+
+
+/**
+ * Represents column data structure
+ */
+interface Column {
+  /**
+   * Status data
+   */
+  status: Status | { _id: string, label: string },
+
+  /**
+   * Tasks that have corresponding status
+   */
+  tasks: Task[]
+}
+
+/**
+ * ProjectBoardView component
+ */
+const ProjectBoardView: React.FC<Props> = () => {
+  const currentProject = useStore($selectedProject);
+  const tasksList = useStore($tasks);
+
+  const onDragEnd = async (result: DropResult): Promise<void> => {
+    const { destination, source, draggableId } = result;
+
+    if (!destination) {
+      return;
+    }
+
+    if (destination.droppableId === source.droppableId && destination.index === source.index) {
+      // Task stays at the same place
+
+      return;
+    }
+
+
+    const updateParams: ChangeTaskStatusPayload = {
```

Comment:
Same problem. High coupling of different levels. Everyone depends on `ChangeTaskStatusPayload`. It should only be the transport-level interface.

UI could just pass `taskId`, `newStatusId` and `order` (or `newIndex`) to the Service level. Service will decide how to pass it to the backend.